src/System.Private.CoreLib/src/System/Threading/Tasks/DebuggerSupport.WinRT.cs

```diff
@@ -0,0 +1,100 @@
+        public static void TraceOperationCreation(CausalityTraceLevel traceLevel, Task task, String operationName, ulong relatedContext)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationCreation(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), operationName, relatedContext);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceOperationCompletion(CausalityTraceLevel traceLevel, Task task, AsyncStatus status)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationCompletion(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), status);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceOperationRelation(CausalityTraceLevel traceLevel, Task task, CausalityRelation relation)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceOperationRelation(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), relation);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceSynchronousWorkStart(CausalityTraceLevel traceLevel, Task task, CausalitySynchronousWork work)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceSynchronousWorkStart(traceLevel, CausalitySource.Library, _platformId, task.OperationId(), work);
+            }
+        }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static void TraceSynchronousWorkCompletion(CausalityTraceLevel traceLevel, CausalitySynchronousWork work)
+        {
+            if (LoggingOn)
+            {
+                WinRTInterop.Callbacks.TraceSynchronousWorkCompletion(traceLevel, CausalitySource.Library, work);
+            }
+        }
+
+        private static ulong OperationId(this Task task)
+        {
+            // The contract with the debugger is that the operation ID contains the appdomain ID in the upper 32 bits and Task.Id in the lower 32 bits.
+            // Project N does not have "appdomains" so we'll put a fixed "id" of 0x00000001 in the upper 32 bits.
+            return 0x0000000100000000LU | (ulong)(uint)(task.Id);
+        }
+
+        static DebuggerSupport()
+        {
+            _loggingOn = false;
+            WinRTInterop.Callbacks.InitTracingStatusChanged(
+                delegate (bool loggingOn)
+                {
+                    _loggingOn = loggingOn;
+                }
+            );
+        }
+
+        private static bool _loggingOn;
+
+        // {4B0171A6-F3D0-41A0-9B33-02550652B995} - Guid that marks our causality events as "Coming from the BCL."
+        private static readonly Guid _platformId = new Guid(0x4B0171A6, unchecked((short)0xF3D0), 0x41A0, 0x9B, 0x33, 0x02, 0x55, 0x06, 0x52, 0xB9, 0x95);
```

Comment:
Nit: I realize this is just a copy of existing code, but these static fields should be prefixed with `s_` instead of `_` (if this repo is following the corefx coding style).